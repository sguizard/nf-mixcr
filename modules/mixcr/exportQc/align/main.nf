process MIXCR_EXPORTQC_ALIGN {
    container 'ghcr.io/milaboratory/mixcr/mixcr:4.6.0'

    input:
    val(study) 
    tuple val(id), path(clns)
    path(library)

    output:
    path("*_exportQC_align.pdf"), emit: qc_align_pdf
    path("*_exportQC_align.png"), emit: qc_align_png
    path "versions.yml"         , emit: versions

    script:
    def args   = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: id

    """
    mixcr exportQc align *.clns ${study}_exportQC_align.pdf
    mixcr exportQc align *.clns ${study}_exportQC_align.png

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        MiXCR: \$( mixcr --version | grep -e 'MiXCR' | sed 's/MiXCR v//' | sed 's/ .\\+//' )
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: id
    """
    touch ${study}_exportQC_align.pdf
    touch ${study}_exportQC_align.png

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        MiXCR: \$( mixcr --version | grep -e 'MiXCR' | sed 's/MiXCR v//' | sed 's/ .\\+//' )
    END_VERSIONS
    """
}
