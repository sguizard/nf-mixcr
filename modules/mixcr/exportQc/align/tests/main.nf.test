nextflow_process {

    name "Test Process MIXCR_EXPORTQC_ALIGN"
    script "modules/mixcr/exportQc/align/main.nf"
    process "MIXCR_EXPORTQC_ALIGN"

    tag "mixcr"
    tag "mixcr/exportQC/align"

    setup {
        run("MIXCR_ANALYZE") {
            script "../../../analyze/main.nf"
            config "../../../analyze/tests/nextflow.config"
            process {
                """
                input[0] = 
                [
                    ['id':'TCR_cat'],
                    [
                        file("${projectDir}/test-datasets/23570MA0002L01_1_20k.fastq.gz"),
                        file("${projectDir}/test-datasets/23570MA0002L01_2_20k.fastq.gz")
                    ]
                ]
                input[1] = 'generic-amplicon-with-umi'
                input[2] = file("${projectDir}/test-datasets/imgt.202312-3.sv8.json.gz")
                """
            }
        }
    }

    test("MIXCR_EXPORTQC_ALIGN Compute align qc") {

        when {
            process {
                """
                input[0] = "TCR_cats"
                input[1] = MIXCR_ANALYZE.out.clns.collect()
                input[2] = file("${projectDir}/test-datasets/imgt.202312-3.sv8.json.gz")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
