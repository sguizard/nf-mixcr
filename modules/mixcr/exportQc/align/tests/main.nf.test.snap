{
    "MIXCR_EXPORTQC_ALIGN Compute align qc": {
        "content": [
            {
                "0": [
                    "TCR_cats_exportQC_align.pdf:md5,fd074615ab1b127628c31367d6b065f2"
                ],
                "1": [
                    "TCR_cats_exportQC_align.png:md5,9579953b2e36b1900bd410448cb627aa"
                ],
                "2": [
                    "versions.yml:md5,870f4bf97a0db96b27903f14a04d0976"
                ],
                "qc_align_pdf": [
                    "TCR_cats_exportQC_align.pdf:md5,fd074615ab1b127628c31367d6b065f2"
                ],
                "qc_align_png": [
                    "TCR_cats_exportQC_align.png:md5,9579953b2e36b1900bd410448cb627aa"
                ],
                "versions": [
                    "versions.yml:md5,870f4bf97a0db96b27903f14a04d0976"
                ]
            }
        ],
        "timestamp": "2024-01-22T12:02:58.356037"
    }
}